### PR TITLE
Add pytest to github workflow

### PR DESCRIPTION
**Pull request by thomasridd**

### CODE REVIEW

File: .github/workflows/python-app.yml [hunk 1]
```diff
@@ -23,7 +23,19 @@ jobs:
...
       run: |
         python -m pip install --upgrade pip
-        pip install flake8
+        pip install flake8 pytest
+        pip install -r requirements-dev.txt
     - name: Lint with flake8
       run: |
-        flake8 . --count --show-source --statistics
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
```
Comment (sanisc) [inline .github/workflows/python-app.yml:31]: there is a .flake8 file in the repo so are these settings needed?

Comment (thomasridd) [reply .github/workflows/python-app.yml]: you're right - if it ain't broke

reverted that

File: .github/workflows/python-app.yml [hunk 2]
```diff
@@ -23,7 +23,16 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install flake8
+        pip install flake8 pytest
```
Comment (sh31) [inline .github/workflows/python-app.yml:26]: Pytest is already included in requirements-dev.txt

Comment (sh31) [reply .github/workflows/python-app.yml:26]: Should flake8 go in there too?

Comment (sanisc) [reply .github/workflows/python-app.yml:26]: might as well add it in makes sense to

Comment (thomasridd) [reply .github/workflows/python-app.yml:26]: cool - I'll remove it

Comment (sanisc) [reply .github/workflows/python-app.yml]: can we added flake8 to dequirements-dev.txt makes sense to @thomasridd

Comment (thomasridd) [reply .github/workflows/python-app.yml]: why not - makes it more explicit about the version of the linter we are using too

**Review by sanisc — APPROVED** [review]

**Pull request merged**